lib/NetPacksLobby.h

```diff
@@ -0,0 +1,311 @@
+	bool checkClientPermissions(CVCMIServer * srv) const;
+	bool applyOnServer(CVCMIServer * srv);
+	void applyOnServerAfterAnnounce(CVCMIServer * srv);
+	bool applyOnLobbyHandler(CServerHandler * handler);
+	void applyOnLobbyScreen(CLobbyScreen * lobby, CServerHandler * handler);
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+		h & initializedStartInfo;
+	}
+};
+
+struct LobbyChangeHost : public CLobbyPackToPropagate
+{
+	int newHostConnectionId;
+
+	LobbyChangeHost() : newHostConnectionId(-1) {}
+	bool checkClientPermissions(CVCMIServer * srv) const;
+	bool applyOnServer(CVCMIServer * srv);
+	bool applyOnServerAfterAnnounce(CVCMIServer * srv);
+
+	template <typename Handler> void serialize(Handler & h, const int version)
+	{
+		h & newHostConnectionId;
+	}
+};
+
+struct LobbyUpdateState : public CLobbyPackToPropagate
+{
+	LobbyState state;
+	bool hostChanged; // Used on client-side only
+
+	LobbyUpdateState() : hostChanged(false) {}
+	bool applyOnLobbyHandler(CServerHandler * handler);
+	void applyOnLobbyScreen(CLobbyScreen * lobby, CServerHandler * handler);
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+		h & state;
+	}
+};
+
+struct LobbySetMap : public CLobbyPackToServer
+{
+	std::shared_ptr<CMapInfo> mapInfo;
+	std::shared_ptr<CMapGenOptions> mapGenOpts;
+
+	LobbySetMap() : mapInfo(nullptr), mapGenOpts(nullptr) {}
+	bool applyOnServer(CVCMIServer * srv);
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+		h & mapInfo;
+		h & mapGenOpts;
+	}
+};
+
+struct LobbySetCampaign : public CLobbyPackToServer
+{
+	std::shared_ptr<CCampaignState> ourCampaign;
+
+	LobbySetCampaign() {}
+	bool applyOnServer(CVCMIServer * srv);
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+		h & ourCampaign;
+	}
+};
+
+struct LobbySetCampaignMap : public CLobbyPackToServer
+{
+	int mapId;
+
+	LobbySetCampaignMap() : mapId(-1) {}
+	bool applyOnServer(CVCMIServer * srv);
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+		h & mapId;
+	}
+};
+
+struct LobbySetCampaignBonus : public CLobbyPackToServer
+{
+	int bonusId;
+
+	LobbySetCampaignBonus() : bonusId(-1) {}
+	bool applyOnServer(CVCMIServer * srv);
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+		h & bonusId;
+	}
+};
+
+struct LobbyChangePlayerOption : public CLobbyPackToServer
+{
+	enum EWhat : ui8 {UNKNOWN, TOWN, HERO, BONUS};
+	ui8 what;
+	si8 direction; //-1 or +1
+	PlayerColor color;
+
+	LobbyChangePlayerOption() : what(UNKNOWN), direction(0), color(PlayerColor::CANNOT_DETERMINE) {}
+	bool checkClientPermissions(CVCMIServer * srv) const;
+	bool applyOnServer(CVCMIServer * srv);
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+		h & what;
+		h & direction;
+		h & color;
+	}
+};
+
+struct LobbySetPlayer : public CLobbyPackToServer
+{
+	PlayerColor clickedColor;
```

Comment:
rename to smth like `selectedColor`

Reply:
I will intentionally keep this one like that until I actually fix all mess of how players positions actually managed by lobby backend code. Since I just moved code that was applied on client to the server it's messy, but since it's working I want to work on it after everything else.